src/NuGet.Clients/NuGet.CommandLine/Commands/ProjectFactory.cs

```diff
@@ -10,16 +10,19 @@
 using System.Xml.Linq;
 using NuGet.Common;
 using NuGet.Configuration;
+using NuGet.Versioning;
 
 namespace NuGet.CommandLine
 {
     using Microsoft.Build.Evaluation;
     using Microsoft.Build.Execution;
-
+    using NuGet.Frameworks;
+    using NuGet.Packaging;
+    using NuGet.Packaging.Core;
     using Console = System.Console;
 
     [SuppressMessage("Microsoft.Maintainability", "CA1506:AvoidExcessiveClassCoupling")]
-    public class ProjectFactory : MSBuildUser, IPropertyProvider
+    public class ProjectFactory : MSBuildUser
     {
```

Comment:
I wouldn't say too far, but it was more than I wanted to bite off. I just checked it out, and there are 136 build errors if I try to switch that file over to not using NuGet.Core.dll (v2), including usage of quite a few classes that weren't ported over.